Gems/Multiplayer/Code/Source/Components/NetworkRigidBodyComponent.cpp

```diff
@@ -119,11 +134,24 @@ namespace Multiplayer
 
     void NetworkRigidBodyComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
     {
-        GetParent().m_physicsRigidBodyComponent->SetKinematic(false);
+        // During activation the simulated bodies are not created yet.
+        // Connect to RigidBodyNotificationBus to listen when it's enabled after creation.
+        Physics::RigidBodyNotificationBus::Handler::BusConnect(GetEntityId());
+    }
+
+    void NetworkRigidBodyComponentController::OnPhysicsEnabled(const AZ::EntityId& entityId)
+    {
+        Physics::RigidBodyNotificationBus::Handler::BusDisconnect();
+
+        m_physicsRigidBodyComponent = Physics::RigidBodyRequestBus::FindFirstHandler(entityId);
+        AZ_Assert(m_physicsRigidBodyComponent, "Physics Rigid Body is required on entity %s", GetEntity()->GetName().c_str());
+
+        m_physicsRigidBodyComponent->SetKinematic(false);
+
 #if AZ_TRAIT_SERVER
         if (IsNetEntityRoleAuthority())
         {
-            if (AzPhysics::RigidBody* rigidBody = GetParent().m_physicsRigidBodyComponent->GetRigidBody())
+            if (AzPhysics::RigidBody* rigidBody = m_physicsRigidBodyComponent->GetRigidBody())
```

Comment:
What was GetParent doing before and why is it no longer necessary?

Reply:
Good question. Unfortunately the controller cannot use the parent's `m_physicsRigidBodyComponent` member because when it's accessed it's not guaranteed it's been initialized yet (it's not guaranteed that `RigidBodyNotificationBus` will call the parent first and then the controller). So that's why now the controller has its own `m_physicsRigidBodyComponent` member.

Reply:
Cool, thanks for the explanation.